src/util/misc/matrix.ts

```diff
@@ -84,6 +82,25 @@ export const multiplyTransformMatrices = (
     is2x2 ? 0 : a[1] * b[4] + a[3] * b[5] + a[5],
   ] as TMat2D;
 
+/**
+ * Multiplies {@link matrices} such that a matrix defines the plane for the rest of the matrices **after** it
+ *
+ * `multiplyTransformMatrixChain([A, B, C, D])` is equivalent to `A(B(C(D)))`
+ *
+ * @param matrices an array of matrices
+ * @param [is2x2] flag to multiply matrices as 2x2 matrices
+ * @returns the multiplication product
+ */
+export const multiplyTransformMatrices2 = (
+  matrices: (TMat2D | undefined | null | false)[],
+  is2x2?: boolean
+) =>
+  matrices.reduceRight(
+    (product: TMat2D, curr) =>
+      curr ? multiplyTransformMatrices(curr, product, is2x2) : product,
+    iMatrix
+  );
```

Comment:
maybe i don't understan reduceRigth but it seems to me we are inverting the order of multiplication of matrices.

Reply:
This is the right order. 
All tests pass.
https://en.wikipedia.org/wiki/Matrix_multiplication#Associativity

Notice the args order passed to `multiplyTransformMatrices`

Reply:
i know the result doesn't change because i know you know math. ( on top of that i know that property too ).
I opened the article anyway to see if maybe there was explained a mathematical property that was the reason for this change, but i couldn't find one.

So i don't know why you think that is the most correct order.
Why do you think is important to change it? I always have express preferences for don't touch what is not getting measurably better and i think you know it by now.

( apart that given the flawed float system of js, changing mutliplication order of floats ( the one inside the matrices ) isn't exactly the same thing. It could be, but sometimes it won't be)
( and apart that most of developers, inclding me, when they see reduceRight, they have to go and check how ti differs from reduce )

Reply:
This is how it is done mathmatically
Because of [Associativity](https://en.wikipedia.org/wiki/Matrix_multiplication#Associativity) it doesn't matter. I wasn't sure at the time of writing but now I am.
So it can be turned into reduce but IMO this is the correct manner.

Reply:
added a test and renamed

Reply:
I will better explain.
Without [Associativity](https://en.wikipedia.org/wiki/Matrix_multiplication#Associativity) it would matter.
Meaning, that using reduce multiplies left to right and is correct only because of [Associativity](https://en.wikipedia.org/wiki/Matrix_multiplication#Associativity). 
Multiplying RTL is how matrix multiplication is defined

Reply:
It is confusing

Reply:
after much thinking i will write you what i think.
I think that even removed any personal preference, the rule of no gain no change should be easy to agree with for everyone.
There is no gain here ( just minor losses ) ( i can't even find references about one order being more canon that the other ) and just the will to have this change.
This was hours ago, now you take this reduceRight and you keep it, my time and mood is not worth this change.
When i get to this point i the never want to talk about it anymore so please help me there.